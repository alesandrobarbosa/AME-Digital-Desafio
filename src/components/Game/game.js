import React, { Component } from 'react';
import swapi from '../../api/swapi';

import Loading from '../Loading/loading';
import ErrorPage from '../Error/error'

import './game.scss';

class Game extends Component {

    constructor(props) {
        super(props);
        this.state = {
            data: '',
            loading: '',
            error: false,
            films: 0
        }

        this.getData = this.getData.bind(this);

    }

    getData() {
        let idPlanet = Math.floor(Math.random() * 61 + 1);
        this.setState({ loading: true })

        fetch(swapi + `planets/${idPlanet}/`)
            .then(data => data.json())
            .then(data => {
                this.setState({ data: data })
                this.setState({ loading: false })
                this.setState({ films: data.films.length })
            })
            .catch(err => {
                this.setState({ error: true })
            })
    }

    componentDidMount() {
        this.getData();

    }

    handleError() {
        if (this.state.error === true) {
            return <ErrorPage />
        }
    }
    

    render() {
        this.handleError();

        return (
            this.state.loading ?
                <Loading />
                :
                <div className="card">
                    <div className="card__header">
                        <h1>{this.state.data.name}</h1>
                    </div>
                    <div className="card__main">
                        <ul className="card__main--list">
                            <li>
                                <span>Population</span>
                                {this.state.data.population}
                            </li>
                            <li>
                                <span>Climate</span>
                                {this.state.data.climate}
                            </li>
                            <li>
                                <span>Terrain</span>
                                {this.state.data.terrain}
                            </li>
                            <li>
                                <span>Featured in Films</span>
                                {this.state.films}
                            </li>
                        </ul>
                    </div>
                    <div className="card__next">
                        <input className="card__next--btn" type="button" value="Next" onClick={this.getData} />
                    </div>
                </div>

        )
    }
}

export default Game;