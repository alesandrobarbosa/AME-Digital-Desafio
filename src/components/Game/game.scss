@import "../../view/styles/styles.scss";

.card {
  margin: auto;
  display: inline-block;
  padding: 20px;
  font-family: $fontStyle;
  background: $cardBgColor;
  color: $cardTxtColor;
  border: 1px solid #ffffff;
  border-radius: 10px;
  box-shadow: 0px 0px 15px 2px;

  &__header {
    text-align: center;
    border-style: dashed;
    border-left: 0;
    border-right: 0;
    h1 {
      line-height: 1.6em;
      font-size: 1.4em;
    }
  }

  &__main {
    max-width: 320px;
    &--list {
      font-size: 0.8em;
      list-style: none;
      padding: 20px;
      max-width: 320px;
      li {
        margin: 20px 0;
        line-height: 1.4em;
        text-align: center;
        border-bottom: 2px solid;
        border-top: 0;
        border-left: 0;
        border-right: 0;
        border-style: dotted;
        padding-bottom: 20px;
        span {
          color: $cardLiTitleColor;
          display: block;
          margin-bottom: 10px;
          font-size: 1.1em;
        }
      }
    }
  }

  &__next {
    &--btn {
      padding: 15px 60px;
      background: #ffffff;
      display: block;
      margin: auto;
      font-family: $fontStyle;
      font-size: 1.2em;
    }
  }
}
