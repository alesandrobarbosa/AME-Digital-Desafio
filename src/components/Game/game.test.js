import React from 'react';
import fetchMock from 'fetch-mock';
import fetch from 'node-fetch'
import renderer from 'react-test-renderer';
import Game from './game';

const {Response} = jest.requireActual('node-fetch');

describe('#GAME COMPONENT ==>', () => {

    it('Game component renders without crashing', () => {
        const component = renderer.create(<Game></Game>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('Fetch request works fine', () => {
        
        fetchMock.mock('*', 200)
        const rendered = renderer.create(<Game />).toJSON()
        expect(rendered).toMatchSnapshot();

    });

})


