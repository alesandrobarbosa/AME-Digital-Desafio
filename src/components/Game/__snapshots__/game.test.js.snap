// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#GAME COMPONENT ==> Fetch should request without crashing 1`] = `
<div
  className="container"
>
  <img
    alt="Loading..."
    className="container__loader"
    src="loader.png"
  />
  Loading...
</div>
`;

exports[`#GAME COMPONENT ==> Game component should renders without crashing 1`] = `
<div
  className="container"
>
  <img
    alt="Loading..."
    className="container__loader"
    src="loader.png"
  />
  Loading...
</div>
`;
